custom/icds_reports/utils/__init__.py

```diff
@@ -1590,3 +1593,25 @@ def get_datatables_ordering_info(request):
 
 def phone_number_function(x):
     return "+{0}{1}".format('' if str(x).startswith('91') else '91', x) if x else x
+
+
+def filter_cas_data_export(export_file, location):
+    loc_type = location.location_type.name
+    with TransientTempfile() as path:
+        with open(path, 'wb') as temp_file:
+            temp_file.write(export_file.get_file_from_blobdb().read())
+        with open(path, 'r') as temp_file:
+            fd, export_file_path = mkstemp()
+            csv_file = os.fdopen(fd, 'w')
+            reader = csv.reader(temp_file)
+            writer = csv.writer(csv_file)
+            headers = next(reader)
+            for i, header in enumerate(headers):
+                if header == f'{loc_type}_name':
+                    index = i
+                    break
+            writer.writerow(headers)
+            for row in reader:
+                if row[index] == location.name:
```

Comment:
I think this will crash if it can't find the header in the loop above since it won't be defined. that may be desired, but it might be better to fail with a clearer exception

Reply:
I think this is the only feedback you missed and is probably worth just quickly initializing/checking the variable and raising a clearer error if it isn't found?